toqito/channels/tests/test_pauli_channel.py

```diff
@@ -0,0 +1,167 @@
+"""Test pauli channel."""
+
+import re
+
+import numpy as np
+import pytest
+from scipy.linalg import eigvalsh
+
+from toqito.channels import partial_trace, pauli_channel
+
+
+@pytest.mark.parametrize(
+    "in_p",
+    [
+        # Negative probability.
+        ([-0.1, 1.1]),
+        # Incorrect length.
+        ([0.2, 0.2, 0.2, 0.2, 0.2]),
+        # Out of range probabilities.
+        ([1.1, 0.1]),
+    ],
+)
+def test_pauli_channel_valid_probability(in_p):
+    """Test probability vector validation with various invalid inputs."""
+    with pytest.raises(
+        ValueError,
+        match=(
+            re.escape("Probabilities must be non-negative and sum to 1.")
+            + "|"
+            + re.escape("The length of the probability vector must be 4^q for some integer q (number of qubits).")
+        ),
+    ):
+        pauli_channel(prob=in_p)
+
+
+@pytest.mark.parametrize(
+    "num_qubits,expected_dim",
+    [
+        # Single qubit.
+        (1, 4),
+        # Two qubits.
+        (2, 16),
+        # Three qubits.
+        (3, 64),
+    ],
+)
+def test_pauli_channel_dimensions(num_qubits, expected_dim):
+    """Test Pauli channel generation for different numbers of qubits."""
+    p = np.ones(expected_dim) / expected_dim
+    Phi = pauli_channel(prob=p)
+    assert Phi.shape == (expected_dim, expected_dim), f"Incorrect matrix dimensions for {num_qubits} qubits"

```

Comment:
We don't need these strings. 
```suggestion
    assert Phi.shape == (expected_dim, expected_dim)

```